### ISSUE TITLE
Block palettes for colorblindness

### ISSUE THREAD

**Issue by DNin01** [ScratchAddons/ScratchAddons#5956]
### Why this would be helpful

For those who are colorblind, it can be difficult to tell block categories apart. I’ve memorized some blocks by using their color and size to distinguish them apart from other blocks so I don’t always have to read them. You can also use colors to determine what kind of operation a block will do. For colorblind users, that can be more difficult, making recognizing blocks not impossible but less convenient.

### My suggestion

Add block color presets that accommodate for Protanopia, Deuteranopia and Tritanopia to `editor-theme3`.

I’ve already created these presets and will submit a PR if others are interested.

### Possible alternatives

Let Scratch implement it themselves, which may take longer but cover a wider range of users.

**Comment by Samq64** [comment]
`block-palette-icons` should at least partly solve that. Maybe a toggle to also add the icons to the block could also be added?

**Comment by WorldLanguages** [comment]
Related: #4446

**Comment by DNin01** [comment]
Before I work on this feature, I think it would be a good idea to have a few other people make colorblind themes to compare. If anyone else wants to help, it'd be great if you could share your own themes, probably one for Protanopia and Deuteranopia and one for Tritanopia. Here's mine:
```json
    {
      "name": "Protanopia/Deuteranopia",
      "id": "protanopia",
      "description": "Block colors modified for users with red/green color blindness",
      "values": {
        "motion-color": "#1DBFFF",
        "looks-color": "#572DFF",
        "sounds-color": "#8088FF",
        "events-color": "#B3D702",
        "control-color": "#C7BC1D",
        "sensing-color": "#55ACBE",
        "operators-color": "#049051",
        "data-color": "#4161E0",
        "data-lists-color": "#4D6FB0",
        "custom-color": "#3C007C",
        "Pen-color": "#015811",
        "sa-color": "#015811",
        "comment-color": "#FEF49C",
        "input-color": "#FFFFFF",
        "text": "white"
      }
    },
    {
      "name": "Tritanopia",
      "id": "tritanopia",
      "description": "Block colors modified for users with blue/yellow color blindness",
      "values": {
        "motion-color": "#4C97FF",
        "looks-color": "#5883BB",
        "sounds-color": "#CF63CF",
        "events-color": "#FFBF00",
        "control-color": "#FFAB19",
        "sensing-color": "#7CC4E8",
        "operators-color": "#277C27",
        "data-color": "#ED8504",
        "data-lists-color": "#F25E1C",
        "custom-color": "#C42148",
        "Pen-color": "#00C750",
        "sa-color": "#00C750",
        "comment-color": "#FEF49C",
        "input-color": "#FFFFFF",
        "text": "white"
      }
    }
```

**What they look like for colorblind users (Protanopia, Deuteranopia and Tritanopia respectively):**

![Screenshot 1](https://user-images.githubusercontent.com/106490990/233758010-b02dccb1-7c50-47b4-bbb8-e6a487c37032.png)![Screenshot 2](https://user-images.githubusercontent.com/106490990/233758008-c81b275d-e3b1-4c49-92dd-5ef9bf88fc99.png)![Screenshot 3](https://user-images.githubusercontent.com/106490990/233758011-03759e82-6475-483a-8c02-8ea4011389c6.png)
<sub>Generated with [Coblis](https://www.color-blindness.com/coblis-color-blindness-simulator/)</sub>
<sub>Oh God, if someone could put the screenshots in a dropdown, that would be great</sub>

We'll find which one is liked best and use it in the PR. Sound like a plan?

**Comment by BroJac5246** [comment]
>2](https://user-images.githubusercontent.com/106490990/233758008-c81b275d-e3b1-4c49-92dd-5ef9bf88fc99.png)![Screenshot]


Deuteranomaly looks good, except that variables and lists are hard to tell apart.

**Comment by Samq64** [comment]
> if someone could put the screenshots in a dropdown, that would be great

That can be done with the `<details>` tag since GitHub supports using some HTML in comments.
https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/organizing-information-with-collapsed-sections

**Comment by DNin01** [comment]
> > if someone could put the screenshots in a dropdown, that would be great
> 
> That can be done with the `<details>` tag since GitHub supports using some HTML in comments. https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/organizing-information-with-collapsed-sections

I know.

Wait, you _can_ use Markdown in the details tags? The last time I tried that I was having trouble adding code blocks and images.

**Comment by mybearworld** [comment]
> Wait, you _can_ use Markdown in the details tags? The last time I tried that I was having trouble adding code blocks and images.

You have to add a new line after the summary tag.
<details><summary>One new line</summary>
```js
// As you can see, this doesn't work
```
</details>
<details><summary>Two new lines</summary>

```js
// As you can see, this does work
```
</details>

**Comment by WorldLanguages** [comment]
> Protanopia, Deuteranopia and Tritanopia

Would be nice if we could consult a specialist, or at least people that are diagnosed with these color vision differences.

My main concern when these three types of colorblindness were brought up at #4446 is that we probably don't have enough knowledge or information to provide a set of accessible colors that is actually useful/ideal for each type of colorblindness.

**Comment by Secret-chest** [comment]
I think we should try a new universal preset that tries to use saturation as well as shade to discriminate colours.

**Comment by DNin01** [comment]
> I think we should try a new universal preset that tries to use saturation as well as shade to discriminate colours.

That would require less research for sure.

**Comment by WorldLanguages** [comment]
If anyone creates a preset and can defend why it's good, and there's enough consensus, it can be added.

**Comment by medlans** [comment]
I made these and got feedback for the blocks
https://scratch.mit.edu/discuss/post/8072319/
https://scratch.mit.edu/discuss/post/8072378/
https://scratch.mit.edu/discuss/topic/694496/?page=31#post-8081404
